### PR TITLE
Add the submit feedback UI

### LINKED ISSUE #493: Settings: Implement Submit Feedback
## Description
This will be presented with a push navigation instead of sheet presentation. But instead of a back button there will be a `cancel` button. Also a `save` button on the other side. Check designs.

[iOS Designs page.](https://toggl.invisionapp.com/d/main?origin=v7#/projects/prototypes/19800511) [Android Designs page.](https://toggl.invisionapp.com/d/main?origin=v7#/projects/prototypes/19932960) [Notion page.](https://www.notion.so/toggl/Settings)

## Dependencies
`Settings: Add settings view` #481
## Definition of done
- [ ] Implement the logic
- [ ] implement the UI

### PR DESCRIPTION
**Pull request by dshr**
## Info
This PR adds the submit feedback UI and hooks it up to the architecture.

### Relationships
Closes #493

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Commit by commit! There's a weird warning happening after the feedback result snackbar disappears that I couldn't pinpoint — any suggestions are welcome: 
```
Reentrancy anomaly was detected.
  > Debugging: To debug this issue you can set a breakpoint in /Users/desislavhristov/Work/toggl-ios/Pods/RxSwift/RxSwift/Rx.swift:96 and observe the call stack.
  > Problem: This behavior is breaking the observable sequence grammar. `next (error | completed)?`
    This behavior breaks the grammar because there is overlapping between sequence events.
    Observable sequence is trying to send an event before sending of previous event has finished.
  > Interpretation: This could mean that there is some kind of unexpected cyclic dependency in your code,
    or that the system is not behaving in the expected way.
  > Remedy: If this is the expected behavior this message can be suppressed by adding `.observeOn(MainScheduler.asyncInstance)`
    or by enqueuing sequence events in some other way.
```

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones
- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Go for it!

### CODE REVIEW

**Review by simonrozsival — CHANGES_REQUESTED** [review]
When the snackbar is hidden after the delay, I'm getting this error/warning in the console:

```
>>>>>>>>>>>>>>
Action: MessageEntered: Asd asd Assad’s asd asd asd asd asd
>>>>>>>>>>>>>>
Action: DoneButtonTapped
>>>>>>>>>>>>>>
Action: SetSendFeedbackError The operation couldn’t be completed. (Networking.NetworkingError error 2.)
>>>>>>>>>>>>>>
Action: SendFeedbackResultSeen
>>>>>>>>>>>>>>
⚠️ Reentrancy anomaly was detected.
  > Debugging: To debug this issue you can set a breakpoint in /Users/simonrozsival/Projects/github/toggl/ios/Pods/RxSwift/RxSwift/Rx.swift:96 and observe the call stack.
  > Problem: This behavior is breaking the observable sequence grammar. `next (error | completed)?`
    This behavior breaks the grammar because there is overlapping between sequence events.
    Observable sequence is trying to send an event before sending of previous event has finished.
  > Interpretation: This could mean that there is some kind of unexpected cyclic dependency in your code,
    or that the system is not behaving in the expected way.
  > Remedy: If this is the expected behavior this message can be suppressed by adding `.observeOn(MainScheduler.asyncInstance)`
    or by enqueuing sequence events in some other way.
```

I suppose that shouldn't happen but I'm not sure what exactly causes it (the delayed action is dispatched from a different thread maybe? I can't see any cyclic dependency in the code and sending two actions in parallel could explain why there's a race..)

File: Modules/Settings/Settings/SubmitFeedback/Views/SubmitFeedbackViewController.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    
+    public override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        unregisterFromKeyboardEvents()
+        disposeBag = DisposeBag()
+    }
+    
+    @objc private func cancel() {
+        let optionMenu = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
```
Comment (sofokli1) [inline Modules/Settings/Settings/SubmitFeedback/Views/SubmitFeedbackViewController.swift:62]: canceling without typing should dismiss without action view

**Review by sofokli1 — CHANGES_REQUESTED** [review]
overall looks ok just some small stuff

**Comment by sofokli1** [comment]
also looked into the bug you have but couldn't find why is happening

**Review by simonrozsival — APPROVED** [review]
The console warning/error is gone, the behavior of the screen seems to be working just fine, the UI looks OK.

File: Modules/Architecture/Architecture/Core/Effect.swift [hunk 2]
```diff
@@ -178,7 +178,6 @@ extension Effect {
             cancellationDisposable = AnyDisposable(
                 Disposables.create {
                     cancellablesLock.sync {
-                        subject.onCompleted()
```
Comment (sofokli1) [inline Modules/Architecture/Architecture/Core/Effect.swift:181]: @Odrakir what do you think about this?

Comment (dshr) [reply Modules/Architecture/Architecture/Core/Effect.swift:181]: He's away until the 28th

Comment (sofokli1) [reply Modules/Architecture/Architecture/Core/Effect.swift:181]: I left this as a placeholder for him to check when he is back

**Review by sofokli1 — APPROVED** [review]

**Pull request merged**